src/basics.md

```diff
@@ -326,6 +327,80 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-parse-subprocess-input]: #ex-parse-subprocess-input
+<a name="ex-parse-subprocess-input"></a>
+## Run an external command passing it stdin and check for an error code
+
+[![regex-badge]][regex] [![cat-os-badge]][cat-os] [![cat-text-processing-badge]][cat-text-processing]
+
+Opens the `python` interpreter using an external [`Command`] and passes it a python statement
+for execution. [`Output`] of said statement is then parsed using [`Regex`] and aggregated for all unique words used in output.
+
+```rust,no_run
+# #[macro_use]
+# extern crate error_chain;
+extern crate regex;
+
+use regex::Regex;
+use std::io::Write;
+use std::ops::Deref;
+use std::process::{Command, Stdio};
+
+# error_chain!{
+#     foreign_links {
+#         Io(std::io::Error);
+#         Regex(regex::Error);
+#         Utf8(std::string::FromUtf8Error);
+#     }
+# }
+
+fn run() -> Result<()> {
+    let mut child = Command::new("python").stdin(Stdio::piped())
+        .stderr(Stdio::piped())
+        .stdout(Stdio::piped())
+        .spawn()?;
+
+    {
+        match child.stdin.as_mut() {
```

Comment:
we actually don't need the surrounding braces as long as we do not really need to use the `stdin` more than once. We can just
```rust
 child.stdin.as_mut()?.write_all(b"import this; exit()")
``` 

As for error handling I would normally use `chain_err` or `map_err` if I would like to add more context. But in this example I would try to keep it as simple as possible so just `?` might suffice.